ui-admin/src/portal/MailingListView.tsx

```diff
@@ -78,33 +83,77 @@ export default function MailingListView({ portalContext, portalEnv }:
   }
   const numSelected = Object.keys(rowSelection).length
 
-
-  useEffect(() => {
+  const loadMailingList = () => {
+    setIsLoading(true)
+    setRowSelection({})
     Api.fetchMailingList(portalContext.portal.shortcode, portalEnv.environmentName).then(result => {
       setContacts(result)
       setIsLoading(false)
     }).catch((e: Error) => {
       alert(`error loading mailing list ${  e}`)
       setIsLoading(false)
     })
+  }
+
+  const performDelete = async () => {
+    const contactsSelected = Object.keys(rowSelection)
+      .filter(key => rowSelection[key])
+      .map(key => contacts[parseInt(key)])
+    try {
+      // this might get gnarly with more than a few entries, but that's okay for now -- this is not expected to be
+      // a heavy-use feature.
+      await Promise.all(
+        contactsSelected.map(contact =>
+          Api.deleteMailingListContact(portalContext.portal.shortcode, portalEnv.environmentName, contact.id)
+        )
+      )
+      Store.addNotification(successNotification(`${contactsSelected.length} entries removed`))
+    } catch {
+      Store.addNotification(failureNotification('Error: some entries could not be removed'))
+    }
+    loadMailingList() // just reload the whole thing to be safe
+    setShowDeleteConfirm(false)
+  }
+
+  useEffect(() => {
+    loadMailingList()
   }, [portalContext.portal.shortcode, portalEnv.environmentName])
+
   return <div className="container p-3">
     <h1 className="h4">Mailing list </h1>
     <LoadingSpinner isLoading={isLoading}>
       <div className="d-flex align-items-center">
         <div>
           {numSelected} of {table.getPreFilteredRowModel().rows.length} selected
         </div>
-        <div>
-          <Button onClick={download}
-            variant="link" disabled={!numSelected}
-            tooltip={numSelected ? 'Download selected contacts' : 'you must select contacts to download'}>
-            Download
-          </Button>
-        </div>
+        <Button onClick={download}
+          variant="secondary" disabled={!numSelected}
+          tooltip={numSelected ? 'Download selected contacts' : 'you must select contacts to download'}>
+          Download
+        </Button>
+        <Button onClick={() => setShowDeleteConfirm(!showDeleteConfirm)}
+          variant="secondary" disabled={!numSelected} className="ms-auto"
+          tooltip={numSelected ? 'Remove selected contacts' : 'you must select contacts to remove'}>
+          Remove
+        </Button>
       </div>
 
       {basicTableLayout(table)}
+      { showDeleteConfirm && <Modal show={true} onHide={() => setShowDeleteConfirm(false)}>
+        <Modal.Body>
+          <div>Do you want to delete the <strong>{ numSelected }</strong> selected entries?</div>
+
+          <div className="pt-3">This operation CANNOT BE UNDONE.</div>
+        </Modal.Body>
+        <Modal.Footer>
+          <button type="button" className="btn btn-primary" onClick={performDelete}>
```

Comment:
```suggestion
          <button type="button" className="btn btn-danger" onClick={performDelete}>
```

Since this is a destructive action, I think it would be appropriate to style this as `danger`